addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/PentairPacket.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pentair.internal;
+
+/**
+ * Generic class for the standard pentair package protocol. Includes helpers to generate checksum and extract key bytes
+ * from packet.
+ *
+ * @author Jeff James - initial contribution
+ *
+ */
+public class PentairPacket {
+    final protected static char[] hexArray = "0123456789ABCDEF".toCharArray();
```

Comment:
Non-preferred order of modifiers correct order is `protected static final` See also: http://stackoverflow.com/questions/16731240/what-is-a-reasonable-order-of-java-modifiers-abstract-final-public-static-e#16732059

+ 6x